x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/CacheService.java

```diff
@@ -144,8 +156,22 @@ protected void doStart() {
 
     @Override
     protected void doStop() {
-        cacheSyncTask.close();
-        cache.invalidateAll();
+        boolean acquired = false;
+        try {
+            acquired = cacheSyncLock.tryLock(cacheSyncStopTimeout.duration(), cacheSyncStopTimeout.timeUnit());
+            if (acquired == false) {
+                logger.warn("failed to acquire cache sync lock in [{}], cache might be partially persisted", cacheSyncStopTimeout);
+            }
+            cacheSyncTask.close();
+            cache.invalidateAll();
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            logger.warn("interrupted while waiting for cache sync lock", e);
```

Comment:
I think we need to also do `cacheSyncTask.close()` and `cache.invalidateAll()` in this case? Possibly better to surround the `tryLock` with a separate try catch for `InterruptedException`.

Reply:
Oh right, I'll surround the `tryLock`